.github/workflows/core_build_publish.yml

```diff
@@ -2,38 +2,71 @@ name: Corelibrary Build & Publish
 
 on:
   push:
-    branches: 
+    branches:
       - 'v[0-9]+.[0-9]+'
   pull_request:
+  workflow_dispatch:
 jobs:
-  build:
+  prepare:
+    name: Prepare & Version
     runs-on: ubuntu-18.04
-    env:
-      DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
-      DOTNET_CLI_TELEMETRY_OPTOUT: 1
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+      publish_artifacts: ${{ steps.version.outputs.publish_artifacts }}
+      publish_nuget: ${{ steps.version.outputs.publish_nuget }}
+      artifacts_name: ${{ steps.version.outputs.artifacts_name }}
     steps:
-    - uses: actions/checkout@v2
-    - name: Setup .NET Core
-      uses: actions/setup-dotnet@v1
-      with:
-        dotnet-version: 3.1.202
     - name: Version
+      id: version
       run: |
         BRANCH=${GITHUB_REF#refs/*/}
         if [[ $BRANCH =~ ^v[0-9]+.[0-9]+$ ]]
         then
           BUILD_NUMBER=$(( GITHUB_RUN_NUMBER + 500 )) # compensate for old jenkins CI
           VERSION="${BRANCH#v}.$BUILD_NUMBER"
-          IS_MASTER_BUILD=1
+          PUBLISH_ARTIFACTS=1
+          ARTIFACTS_NAME="LeanCode.CoreLibrary.$VERSION.zip"
         else
           VERSION="0.0.0"
-          IS_MASTER_BUILD=0
+          PUBLISH_ARTIFACTS=0
+          ARTIFACTS_NAME="<none>"
         fi
-        echo Building on "$BRANCH"
-        echo Building version: "$VERSION"
+        echo $BRANCH
+        echo $VERSION
+        echo "Artifacts will be saved as $ARTIFACTS_NAME" 
         
-        echo "::set-env name=VERSION::${VERSION}"
-        echo "::set-env name=IS_MASTER_BUILD::${IS_MASTER_BUILD}"
+        if [[ $GITHUB_EVENT_NAME == 'workflow_dispatch' ]]
+        then
+          echo "Packages will be published to NuGet"
+          PUBLISH_NUGET=1
+        else
+          PUBLISH_NUGET=0
+        fi
+        
+        if [ $PUBLISH_ARTIFACTS == 0 ] && [ $PUBLISH_NUGET == 1 ]
```

Comment:
I'd prefer `[[` over `[` or `test` like you used above.

Reply:
```suggestion
        if [[ $PUBLISH_ARTIFACTS == 0  &&  $PUBLISH_NUGET == 1 ] ]
```